src/irc.nim

```diff
@@ -142,10 +142,10 @@ proc send*(irc: AsyncIrc, message: string,
   ## quickly to prevent excessive flooding of the IRC server. You can prevent
   ## buffering by specifying ``True`` for the ``sendImmediately`` param.
   if wasBuffered(irc, message, sendImmediately):
-    result = irc.sock.send(message & "\c\L")
-  else:
-    result = newFuture[void]("irc.send")
-    result.complete()
+    if irc.status notin [SockClosed, SockConnecting]:
```

Comment:
Shouldn't this really be an assert? The caller should verify that the socket is writeable, silently ignoring a send could be confusing for the caller.

Reply:
Sure, but the existing code doesn't raise, so in the interests of not breaking anyone...

Reply:
meh, it's not like this is 1.0 :P